packages/template-retail-react-app/app/components/island/README.md

```diff
@@ -0,0 +1,50 @@
+
+In addition to Suspense, [Server Components](https://react.dev/reference/rsc/server-components) are now another official concept in React for addressing the all-or-nothing problem with hydration. However, as the name suggests, Server Components are components that are rendered exclusively on the server and therefore don’t receive any downstream client-side hydration.
+
+## `<Island/>` Component
+
+Unlike Suspense, the `<Island/>` component explicitly **encourages the synchronous rendering of HTML on the server** as much as possible. And instead of delaying the rendering of certain contents, it allows the **explicit delay of the hydration process itself**.
+
+This delay can be achieved using various `hydrateOn` strategies:
+
+1. `hydrateOn={'load'}`: Useful for high-priority UI elements that need to be interactive as quickly as possible. Default behavior if `hydrateOn` is not defined.
+2. `hydrateOn={'idle'}`: Useful for lower-priority UI elements that don’t need to be immediately interactive. Hydration occurs during the [browser’s next available idle period](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback).
+3. `hydrateOn={'visible'}`: Useful for lower-priority UI elements that don’t need to be immediately interactive. Hydration occurs once the component has [entered the user’s viewport](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver). As a result, such an element doesn’t get hydrated at all if the user never sees it.
+4. `hydrateOn={'off'}`: Useful for non-interactive UI elements or lower-priority UI elements to completely suppress the hydration for. On the surface, this behavior overlaps in certain ways with Server Components. However, there are fundamental differences. While server components don’t contribute any JavaScript to what’s delivered to the client, islands are client-oriented constructs, and thus the JavaScript for a non-hydrated section is still fully transmitted. After all, `hydrateOn` can be dynamically updated at any time in case of using a binding property instead of a hard-coded value. An initial value of `off` therefore opens up the possibilities for completely custom hydration triggers. Furthermore, if Server Components aren’t supported by the runtime used, `hydrateOn={'off'}` offers a simple way to at least reduce the hydration overhead for certain non-interactive areas.
+
+> ❗ Ultimately, the approaches of Suspense, Server Components and the `<Island/>` component aren’t mutually exclusive; on the contrary, they can perfectly complement each other.
+
+## `<Island/>` Nesting
```

Comment:
It's thoughtful to call out the nesting 👍 You know someone will be asking this :)